homeassistant/components/ovo_energy/config_flow.py

```diff
@@ -53,7 +62,10 @@ async def async_step_user(self, user_input=None):
             step_id="user", data_schema=USER_SCHEMA, errors=errors
         )
 
-    async def async_step_reauth(self, user_input):
+    async def async_step_reauth(
+        self,
+        user_input: Mapping[str, Any],
```

Comment:
`user_input` should never be None for `async_step_reauth`, so it would be better to split out the logic into a secondary step `reauth_confirm` and adjust the logic accordingly.